lib/matplotlib/widgets.py

```diff
@@ -613,43 +607,94 @@ def _clicked(self, event):
                 self.set_active(i)
                 break
 
-    def set_active(self, index):
+    def set_active(self, index, state=_TOGGLE):
         """
         Directly (de)activate a check button by index.
 
-        *index* is an index into the original label list
-            that this object was constructed with.
-            Raises ValueError if *index* is invalid.
+        Default behaviour is to toggle the state, which can be controlled with
```

Comment:
That's a bit misleading: toggling is not controlled via a bool state.

I would remove this sentence and add the information to the state parameter. See below.